components/Swipe.js

```diff
@@ -0,0 +1,279 @@
+    super(props)
+    this.state = {
+      currentIndex: props.initialIndex,
+      width: 0,
+      scrollLeft: 0,
+      drag: 0
+    }
+
+    autoBind([
+      'setWidth',
+      'handleTouchStart',
+      'handleTouchEnd',
+      'handleTouchMove',
+      'gotoNext',
+      'gotoPrev',
+      'gotoSlide',
+      'onChange',
+      'initLazyLoad',
+      'hasSingleImage',
+      'autoPlay',
+      'pause',
+      'handleClick'
+    ], this)
+  }
+
+  componentDidMount () {
+    this.setWidth()
+    this.initLazyLoad()
+
+    if (this.props.autoPlay) this.autoPlay(this.props)
+
+    if (this.props.responsive) {
+      window.addEventListener('resize', this.setWidth, false)
+    }
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (this.props.autoPlay !== nextProps.autoPlay) {
+      nextProps.autoPlay ? this.autoPlay(nextProps) : this.pause()
+    }
+  }
+
+  setWidth () {
+    if (this.swipeRef) {
+      this.setState({
+        width: this.swipeRef.clientWidth
+      })
+    }
+  }
+
+  handleTouchStart (e) {
+    if (this.hasSingleImage()) return
+    this.clientX = e.touches[0].clientX
+    document.addEventListener('touchmove', this.handleTouchMove, false)
+  }
+
+  handleTouchEnd () {
+    if (this.hasSingleImage()) return
+    const {drag, currentIndex, width} = this.state
+    const {threshold} = this.props
+
+    const initialIndex = currentIndex
+
+    if (Math.abs(drag) > threshold * width) {
+      this.setState(() => {
+        if (drag < 0 && currentIndex > 0) {
+          return {
+            // goto previous slide
+            currentIndex: currentIndex - 1
+          }
+        } else if (drag > 0 && ((currentIndex + 1) < this.props.children.length)) {
+          return {
+            // goto next slide
+            currentIndex: currentIndex + 1
+          }
+        }
+      }, () => {
+        this.onChange(initialIndex)
+      })
+    }
+
+    this.setState({
+      drag: 0
+    })
+
+    document.removeEventListener('touchmove', this.handleTouchMove, false)
+  }
+
+  initLazyLoad () {
+    const {currentIndex} = this.state
+    const {overScan, children} = this.props
+    this[`imageRef${currentIndex}`].load()
+    if (overScan === 1) {
+      if (currentIndex > 0) this[`imageRef${currentIndex - 1}`].load()
+      if (currentIndex + 1 < children.length) this[`imageRef${currentIndex + 1}`].load()
+    }
+  }
+
+  handleTouchMove (e) {
+    const dx = this.clientX - e.touches[0].clientX
+
+    this.setState({
+      drag: dx
+    })
+  }
+
+  gotoPrev () {
+    const {currentIndex} = this.state
+    if (currentIndex > 0) {
+      this.gotoSlide(currentIndex - 1)
+    }
+  }
+
+  gotoNext () {
+    const {currentIndex} = this.state
+    if (currentIndex + 1 < this.props.children.length) {
+      this.gotoSlide(currentIndex + 1)
+    }
+  }
+
+  gotoSlide (i) {
+    const initial = this.state.currentIndex
+    this.setState({
+      currentIndex: i
+    }, () => (this.onChange(initial)))
+  }
+
+  autoPlay (props) {
```

Comment:
Test Case for component with auto play:

When I manually go to the next slide, this.pause() should be called, else after time out, the user will go to the next screen.